staging/src/k8s.io/apiserver/pkg/registry/generic/registry/corrupt_obj_deleter.go

```diff
@@ -107,8 +107,16 @@ func (d *corruptObjectDeleter) Delete(ctx context.Context, name string, deleteVa
 	klog.FromContext(ctx).V(1).Info("Going to perform unsafe object deletion", "object", klog.KRef(genericapirequest.NamespaceValue(ctx), name))
 	out := d.store.NewFunc()
 	storageOpts := storage.DeleteOptions{IgnoreStoreReadError: true}
-	// dropping preconditions, and keeping the admission
-	if err := storageBackend.Delete(ctx, key, out, nil, storage.ValidateObjectFunc(deleteValidation), nil, storageOpts); err != nil {
+	// dropping preconditions, and admission since we don't have the old
+	// object in the cache, neither can it be retrieved from the storage
+	// and decoded into an object successfully.
+	var (
+		preconditions        *storage.Preconditions = nil
+		cachedExistingObject runtime.Object         = nil
+		// rest.ValidateAllObjectFunc will "admit everything"
+		validateDeletion storage.ValidateObjectFunc = rest.ValidateAllObjectFunc
+	)
+	if err := storageBackend.Delete(ctx, key, out, preconditions, validateDeletion, cachedExistingObject, storageOpts); err != nil {
```

Comment:
```suggestion
	// dropping preconditions and admission since we don't have the old object
	// in the cache, neither can it be retrieved from the storage
	// and decoded into an object successfully.
	preconditions        *storage.Preconditions = nil
	cachedExistingObject runtime.Object         = nil
	if err := storageBackend.Delete(ctx, key, out, preconditions, rest.ValidateAllObjectFunc, cachedExistingObject, storageOpts); err != nil {
```

Reply:
less visual noise.